src/index.js

```diff
@@ -64,19 +72,13 @@ export function optimize(data, peakList, options = {}) {
     }
   }
 
-  lmOptions = Object.assign(
-    {
-      damping: 1.5,
-      initialValues: pInit,
-      minValues: pMin,
-      maxValues: pMax,
-      gradientDifference: dt / 10000,
-      maxIterations: 100,
-      errorTolerance: 10e-5,
-    },
-    lmOptions,
-  );
-  let pFit = LM({ x, y }, paramsFunc, lmOptions);
+  let { algorithm, optOptions } = choiceMethod(optimization);
+
+  optOptions.minValues = pMin;
+  optOptions.maxValues = pMax;
+  optOptions.initialValues = pInit;
+
+  let pFit = algorithm({ x, y }, paramsFunc, optOptions);
 
   let { parameterError: error, iterations } = pFit;
   let result = { error, iterations, peaks: new Array(nL) };
```

Comment:
Could you check if it could be an empty array in which you push ? It should be faster